src/vue/forms/TfaForm.vue

```diff
@@ -0,0 +1,237 @@
+          </template>
+          <template v-else>
+            {{ 'tfa-form.next-btn' | globalize }}
+          </template>
+        </button>
+      </div>
+    </template>
+    <template v-else>
+      <p class="tfa-form__hint">
+        {{ 'tfa-form.qr-desc' | globalize }}
+      </p>
+      <key-viewer
+        :value="form.factor.seed"
+        :is-clipboard-shown="false"
+      />
+      <p class="tfa-form__hint">
+        {{ 'tfa-form.secret-code-desc' | globalize }}
+      </p>
+      <clipboard-field
+        :value="form.factor.secret"
+        :label="'tfa-form.secret-lbl' | globalize"
+      />
+      <p class="tfa-form__hint tfa-form__code-hint">
+        {{ 'tfa-form.tfa-code-desc' | globalize }}
+      </p>
+      <div class="app__form-field">
+        <input-field
+          v-model="form.code"
+          @blur="touchField('form.code')"
+          id="tfa-code"
+          :error-message="getFieldErrorMessage('form.code')"
+          :label="'tfa-form.code-lbl' | globalize"
+          :disabled="formMixin.isDisabled"
+        />
+      </div>
+      <div class="app__form-actions">
+        <button
+          v-ripple
+          type="submit"
+          class="tfa-form__btn"
+          :disabled="formMixin.isDisabled"
+        >
+          {{ 'tfa-form.enable-btn' | globalize }}
+        </button>
+      </div>
+    </template>
+  </form>
+</template>
+
+<script>
+import FormMixin from '@/vue/mixins/form.mixin'
+import KeyViewer from '@/vue/common/KeyViewer'
+import ClipboardField from '@/vue/fields/ClipboardField'
+
+import { required, password } from '@validators'
+
+import { ErrorHandler } from '@/js/helpers/error-handler'
+import { Bus } from '@/js/helpers/event-bus'
+
+import { Sdk } from '@/sdk'
+import { errors } from '@tokend/js-sdk'
+
+import { vuexTypes } from '@/vuex'
+import { mapGetters, mapActions } from 'vuex'
+
+const ENABLED_FACTOR_PRIORITY = 1
+const EVENTS = {
+  update: 'update',
+}
+
+export default {
+  name: 'tfa-form',
+  components: {
+    KeyViewer,
+    ClipboardField,
+  },
+  mixins: [FormMixin],
+  data: _ => ({
+    form: {
+      password: '',
+      factor: null,
+      code: '',
+    },
+  }),
+  validations: {
+    form: {
+      password: { required, password },
+      code: { required },
+    },
+  },
+  computed: {
+    ...mapGetters({
+      isTotpEnabled: vuexTypes.isTotpEnabled,
+      totpFactors: vuexTypes.factorsTotp,
+    }),
+  },
+  async created () {
+    try {
+      await this.loadFactors()
+    } catch (e) {
+      ErrorHandler.processWithoutFeedback(e)
+    }
+  },
+  methods: {
+    ...mapActions({
+      loadFactors: vuexTypes.LOAD_FACTORS,
+    }),
+    async toggleFactor () {
+      if (this.$v.form.password.$invalid) return
+      this.disableForm()
+      if (this.isTotpEnabled) {
+        await this.deleteTotpFactor()
+      } else {
+        await this.createTotpFactor()
+      }
+      this.enableForm()
+    },
+    async createTotpFactor () {
+      try {
```

Comment:
@quinn71, again the try-catch hell. created the task: https://app.asana.com/0/894507180698157/1108030165482985/f